### ISSUE TITLE
timing of page load and WebViewJavascriptBridgeReady event

### ISSUE THREAD

**Issue by refractalize** [marcuswestin/WebViewJavascriptBridge#56]
I've experienced problems connecting the page with the bridge in a couple of different instances, and in both cases it was due to the page not loading fast enough to start listening for the `WebViewJavascriptBridgeReady` event.

The problem may occur when you have components that use the bridge that are executed well after the page is initially loaded, for example, I was trying to connect to the bridge in an AngularJS app which has a prolonged page load sequence.

The other problem I saw was that the `webViewDidFinishLoad:` was being fired before all the scripts had loaded. In this case the scripts were at the bottom of the `<body>` element so perhaps UIWebView is considering the body of the page effectively loaded since the scripts don't contribute to the initial rendering. The result was, again, that the `WebViewJavascriptBridgeReady` event was being fired before the page JS could listen to it.

Anyway, long story short, the solution for me was to do something like this:

```
function connectToBridge(fn) {
  if (window.WebViewJavascriptBridge) {
    fn(WebViewJavascriptBridge);
  } else {
    document.addEventListener('WebViewJavascriptBridgeReady', function(ev) {
      fn(ev.bridge);
    });
  }
}

connectToBridge(function (bridge) {
  bridge.init(function (msg) {
      ...
  });
});
```

Essentially, detect that bridge hasn't already loaded before listening for the event.

I wondered if you'd consider putting this on the `README.md`? Might save somebody else some time.

Otherwise, WebViewJavascriptBridge is awesome, thank you!
